libs/execution/include/hpx/execution/executors/thread_execution.hpp

```diff
@@ -205,23 +204,16 @@ namespace hpx { namespace threads {
         typedef hpx::future<vector_result_type> result_future_type;
 
         typedef typename hpx::traits::detail::shared_state_ptr<
-            result_future_type>::type shared_state_type;
+            vector_result_type>::type shared_state_type;
 
         typedef typename std::decay<Future>::type future_type;
 
-        thread_id_type id = hpx::threads::get_self_id();
-        parallel::execution::current_executor exec_current =
-            hpx::threads::get_executor(id);
-
         shared_state_type p =
-            lcos::detail::make_continuation_exec<result_future_type>(
+            lcos::detail::make_continuation_exec<vector_result_type>(
                 std::forward<Future>(predecessor), std::forward<Executor>(exec),
-                [func = std::move(func),
-                    exec_current = std::move(exec_current)](
-                    future_type&& predecessor) mutable -> result_future_type {
-                    return hpx::parallel::execution::async_execute(exec_current,
-                        hpx::util::functional::unwrap{},
-                        func(std::move(predecessor)));
+                [func = std::move(func)](
+                    future_type&& predecessor) mutable -> vector_result_type {
+                    return hpx::util::unwrap(func(std::move(predecessor)));
```

Comment:
This possibly changes the thread that executes the lambda. Are we sure we want that?

Reply:
I'm aware. I'm going by the tests and they seem to accept this. Since this is deprecated functionality I did not investigate this further. Again, this functionality could move to the `thread_executors` module, allowing us to keep the old behaviour

Reply:
I take that back. The `thread_pool_os_executors` test occasionally fails after this change. I've reverted the change and moved all remaining thread executor related files to `thread_executors`.

Note that the changes in this PR modules do slightly break compatibility (so much for that...). In 1.4.1 one could include `hpx/parallel/execution.hpp` and get all the execution related headers (executors, execution, thread execution, and timed execution). This is not possible anymore (well, we could keep that true, but have chosen not to). The catch-all header `hpx/include/parallel_execution.hpp` is immune against these changes. I think these changes are low risk though as most users should be on the newer executors in `hpx::parallel::execution` and not `hpx::threads::executors`. Timed execution I would guess is rare.